doc/whats_new/v1.1.rst

```diff
@@ -34,6 +34,15 @@ random sampling procedures.
   same solution, up to numerical rounding errors, but in general Lloyd's
   algorithm uses much less memory, and it is often faster.
 
+- |Efficiency| :class:`tree.DecisionTreeClassifier`,
+  :class:`tree.DecisionTreeRegressor`,
+  :class:`ensemble.RandomForestClassifier`,
+  :class:`ensemble.RandomForestRegressor`,
+  :class:`ensemble.GradientBoostingClassifier`, and
+  :class:`ensemble.GradientBoostingRegressor` are on average 15% faster may
+  result in a different model due to a different unstable sort algorithm for
+  finding the best split. :pr:`22868` by `Thomas Fan`_.
```

Comment:
Is this phrasing more exact? Should we be more explicit for users to understand the reason for the model change?
```suggestion
- |Efficiency| Fitting :class:`tree.DecisionTreeClassifier`,
  :class:`tree.DecisionTreeRegressor`,
  :class:`ensemble.RandomForestClassifier`,
  :class:`ensemble.RandomForestRegressor`,
  :class:`ensemble.GradientBoostingClassifier`, and
  :class:`ensemble.GradientBoostingRegressor` is on average 15% faster than
  in previous versions thanks to a new sort algorithm to find the best split.
  Models might be a slightly different due to the difference of stability
  of between the previous and the new algorithms. :pr:`22868` by `Thomas Fan`_.
```

Reply:
Since both are unstable I find it weird to speak of "difference of stability".

> Fitted models might be a slightly different due to the difference in the  handling of splits of tied objective values (both the old and the new sorting algorithm are unstable).